src/Raven.Server/Documents/Indexes/Persistence/Lucene/IndexReadOperation.cs

```diff
@@ -49,15 +49,19 @@ public class IndexReadOperation : IndexOperationBase
         private readonly IDisposable _releaseSearcher;
         private readonly IDisposable _releaseReadTransaction;
 
-        public IndexReadOperation(string indexName, IndexType indexType, int maxIndexOutputsPerDocument, int? actualMaxIndexOutputsPerDocument, Dictionary<string, IndexField> fields, LuceneVoronDirectory directory, IndexSearcherHolder searcherHolder, Transaction readTransaction)
+        public IndexReadOperation(string indexName, IndexType indexType,
+            int maxIndexOutputsPerDocument, int? actualMaxIndexOutputsPerDocument,
+            Dictionary<string, IndexField> fields, LuceneVoronDirectory directory,
+            IndexSearcherHolder searcherHolder, Transaction readTransaction, LoggerSetup loggerSetup)
         {
             _analyzer = CreateAnalyzer(() => new LowerCaseKeywordAnalyzer(), fields, forQuerying: true);
             _indexName = indexName;
             _indexType = indexType;
             _actualMaxIndexOutputsPerDocument = actualMaxIndexOutputsPerDocument;
             _maxIndexOutputsPerDocument = maxIndexOutputsPerDocument;
             _releaseReadTransaction = directory.SetTransaction(readTransaction);
-            _releaseSearcher = searcherHolder.GetSearcher(out _searcher);
+            _releaseSearcher = searcherHolder.GetSearcher(out _searcher, loggerSetup);
+            _logger = loggerSetup.GetLogger<IndexReadOperation>("IndexReadOperation");
```

Comment:
You should be passing the db name here, otherwise, we have no idea which db is generating those logs